src/mlx_omni_server/rerank/rerank_service.py

```
@@ -0,0 +1,117 @@
+
+
+class RerankService:
+    """Service for reranking documents using MLX models."""
+
+    def __init__(self):
+        # Map of loaded models for caching
+        self._models: Dict[str, MLXReranker] = {}
+        # Default encoder for token counting
+        try:
+            self._default_tokenizer = tiktoken.get_encoding("cl100k_base")
+        except:
```

Ruff (0.14.6)

Do not use bare `except`

(E722)